console/HAL-1832_Activate_OIDC.adoc

```diff
@@ -0,0 +1,95 @@
+= HAL-1832 Activate OIDC in the console
+:author:            Harald Pehl
+:email:             hpehl@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.redhat.com/browse
+
+== Overview
+
+With the Keycloak OpenID Connect (OIDC) adapter, it was possible to secure the management console using OIDC. When accessing the management console, the user would get redirected to the Keycloak login page, log in with their credentials, and get redirected back to the management console upon successful authentication. It was also possible for the user to log out of the console.
+
+This RFE is to add the ability to secure the management console when using the native support for OIDC. It addresses the steps necessary to configure the resources in `/subsystem=elytron-oidc-client`.
+
+== Issue Metadata
+
+=== Issue
+
+* {issue-base-url}/HAL-1827[HAL-1832]
+
+=== Related Issues
+
+* {issue-base-url}/EAP7-1796[EAP7-1796]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:spriadka@redhat.com[Simon Priadka]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [x] QE
+
+=== Affected Projects or Components
+
+This RFE affects the management console. It depends on the management resources defined by {issue-base-url}/EAP7-1796[EAP7-1796].
+
+=== Other Interested Projects
+
+None
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [ ] OpenShift s2i
+
+* [ ] Bootable jar
+
+== Requirements
+
+Affected UI:: Configuration / Subsystems / Elytron OIDC Client
+Affected Resources:: `/subsystem=elytron-oidc-client`
+
+=== Hard Requirements
+
+The console shows the new `elytron-oidc-client` subsystem. The UI for this subsystem is backed by the https://hal.github.io/documentation/concepts/#applications[model browser] which generates a UI based on the resource descriptions.
+
+The UI makes it possible to configure the necessary resources as described in the Elytron admin guide: `docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Elytron_OIDC_Client.adoc`, section "Securing the management console with OpenID Connect". In particular, this means the console makes it possible to add the following resources:
```

Comment:
Is possible configure RBAC via HAL console?
```
/core-service=management/access=authorization:write-attribute(name=provider,value=rbac)
/core-service=management/access=authorization:write-attribute(name=use-identity-roles,value=true)
```

Reply:
This is actually a good question as the scenario with RBAC `access-control` is valid in this context, see the documented CLI steps from the base RFE: https://github.com/wildfly/wildfly/pull/16856/files#diff-82d49f155de7505904b2e963e7ecb31f2ab37132f5fb93c2e74dddb6d5a7497dR347

Reply:
Yes, it is possible to configure RBAC in the console.

Reply:
Where i can find this setup? Is there also possibility to configure `use-identity-roles`?

Reply:
![CleanShot 2023-06-05 at 10 49 38](https://github.com/wildfly/wildfly-proposals/assets/100611/357f1440-f275-4ba8-ac02-51ae26736026)

To modify the attribute `use-identity-roles` you need to use the generic model browser:
![CleanShot 2023-06-05 at 11 01 54](https://github.com/wildfly/wildfly-proposals/assets/100611/348d4369-9d41-4d48-a033-bdfe463147d8)